VTXdigi/src/VTXdigitizer.cpp

```diff
@@ -126,22 +138,20 @@ StatusCode VTXdigitizer::execute(const EventContext&) const {
 
     // Smear the hit in the local sensor coordinates
     double digiHitLocalPosition[3];
-    if (m_readoutName == "VTXIBCollection" ||
-        m_readoutName == "VTXOBCollection" ||
-        m_readoutName == "VertexBarrelCollection" ||
+    int iLayer = m_decoder->get(cellID, "layer");
```

Comment:
Is this crashing if the field "layer" is not defined in the bit-field encoding? Shall we set the "layer" string key as a Gaudi proprty, check if this exists in the provided readout, and elegantly crash if it does not? Or is this already crashing nicely?

Reply:
No, this does not crash nicely. Declare as key Gaudi property like "  declareProperty("inputSimHits", m_input_sim_hits, "Input sim vertex hit collection name");" ?

Reply:
I've added a check and tested it, let me know if it's fine. I furthermore also added the silicon wrapper digitisation :-)